src/game/map.rs

```diff
@@ -58,10 +63,52 @@ pub fn get_world_size() -> u32 {
 /// See [http://docs.screeps.com/api/#Game.map.isRoomAvailable]
 ///
 /// [http://docs.screeps.com/api/#Game.map.isRoomAvailable]: http://docs.screeps.com/api/#Game.map.isRoomAvailable
+#[deprecated(
+    since = "0.8.0",
+    note = "Game.map.isRoomAvailable is deprecated, use get_room_status instead"
+)]
 pub fn is_room_available(room_name: RoomName) -> bool {
     js_unwrap!(Game.map.isRoomAvailable(@{room_name}))
 }
 
+/// See [http://docs.screeps.com/api/#Game.map.getRoomStatus]
+///
+/// [http://docs.screeps.com/api/#Game.map.getRoomStatus]: http://docs.screeps.com/api/#Game.map.getRoomStatus
+pub fn get_room_status(room_name: RoomName) -> MapRoomStatus {
+    js_unwrap!(Game.map.getRoomStatus(@{room_name}))
+}
+
+/// Represents the availability and respawn/novice state of a room on the map
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MapRoomStatus {
+    #[serde(deserialize_with = "RoomStatus::deserialize_from_str")]
```

Comment:
Not sure you need this. I believe enums can be deserialized from strings out of the box, if you don't use `repr(u8)`, which I think is unnecessary here.